packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -159,30 +155,27 @@ export class ToastQueue<T> {
     if (index >= 0) {
       this.queue[index].onClose?.();
       this.queue.splice(index, 1);
+      let closedToast = this.visibleToasts.find(t => t.key === key);
+      if (closedToast) {
+        closedToast.animation = 'exiting';
+      }
     }
 
-    this.updateVisibleToasts(index);
+    this.updateVisibleToasts('close');
   }
 
   /** Removes a toast from the visible toasts after an exiting animation. */
   remove(key: string) {
-    let index = this.queue.findIndex(t => t.key === key);
-    this.visibleToasts = this.visibleToasts.filter(t => t.key !== key);
-    this.updateVisibleToasts(index);
+    this.queue = this.queue.filter(t => t.key !== key);
+    this.updateVisibleToasts('remove');
   }
```

Comment:
I'm unsure about whether or not to have `this.queue = this.queue.filter(t => t.key !== key);` in here or not. The code works without it assuming the user calls `close` and then `remove` and would allow us to remove the `key` input argument. However, if the user calls just `remove` then we need the aforementioned line to make sure the queue is properly updated and thus `visibleToasts` gets updated as well.

Reply:
Technically it does work, the remove() is just doing some cleanup to the dom by removing the object from the dom.

The close button not closing until you move the mouse was related to close() not remove(), right?

Reply:
Probably good to keep it to be safe.

Reply:
yeah, the extra update to the queue in the `remove` call is to ensure the queue as our source of truth is still up to date in case the user only calls `remove`.  

> The close button not closing until you move the mouse was related to close() not remove(), right?

yeah it previous happened to use since we weren't always making an update to `visibleToast` that would cause a state change

Reply:
I got rid of the extra update to queue in the `remove` call since this was prematurely removing toasts that get pushed out of visibility by higher priority toasts. In RSP the flow that happens when a toast is pushed out of visibility is as follows:
new toast `add` animates in -> toast that is no longer visible gets `close` called by its `onAnimationEnd`